database/sqlite/sqlite_metadata.c

```diff
@@ -0,0 +1,1485 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "sqlite_functions.h"
+#include "sqlite_metadata.h"
+
+extern DICTIONARY *rrdhost_root_index;
+
+// Metadata functions
+#define SQL_DELETE_HOST_LABELS  "DELETE FROM host_label WHERE host_id = @uuid;"
+
+struct query_build {
+    BUFFER *sql;
+    int count;
+    char uuid_str[UUID_STR_LEN];
+};
+
+static int host_label_store_to_sql_callback(const char *name, const char *value, RRDLABEL_SRC ls, void *data) {
+    struct query_build *lb = data;
+    if (unlikely(!lb->count))
+        buffer_sprintf(lb->sql, "INSERT OR REPLACE INTO host_label (host_id, source_type, label_key, label_value, date_created) VALUES ");
```

Comment:
Is there a special reason for `INSERT OR REPLACE` to be passed as argument instead to create a `define` for it?

Reply:
Will add defines